src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -618,32 +628,422 @@ private static void setModuleName(ModuleInputConfiguration.Builder moduleInputCo
+            actualDefaultStr = pathDefaults.get(fullyQualifiedModuleName).get(propertyName);
+        }
+        else if (commonDefaults.containsKey(propertyName)) {
+            actualDefaultStr = commonDefaults.get(propertyName);
+        }
+        else {
+            throw new IllegalStateException("Unable to validate default value for property '"
+            + propertyName + "' in " + fullyQualifiedModuleName);
+        }
+        isDefaultValues(propertyDefaultValue, actualDefaultStr, String.class);
+    }
+
+    /**
+     * Validate default value.
+     *
+     * @param propertyName the property name.
+     * @param propertyDefaultValue the specified default value in the file.
+     * @param fullyQualifiedModuleName the fully qualified module name.
+     * @noinspectionreason IfStatementWithTooManyBranches - complex logic of violation
+     *      parser requires giant if/else
+     */
+    // -@cs[MethodLength] splitting this method is not reasonable
+    // -@cs[ExecutableStatementCount] splitting this method is not reasonable.
+    // -@cs[JavaNCSS] splitting this method is not reasonable.
+    // -@cs[CyclomaticComplexity] splitting this method is not reasonable.
+    private static void defaultValidation(String propertyName,
+                                           String propertyDefaultValue,
+                                           String fullyQualifiedModuleName) {
+        try {
+            final Object checkInstance = createCheckInstance(fullyQualifiedModuleName);
+            final Object actualDefault;
+            final Class<?> propertyType;
+            if ("tokens".equals(propertyName)) {
+                final Method getter = checkInstance.getClass().getMethod("getDefaultTokens");
+                actualDefault = getter.invoke(checkInstance);
+                propertyType = actualDefault.getClass();
+            }
+
+            else {
+                final Field field = checkInstance.getClass().getDeclaredField(propertyName);
+                field.setAccessible(true);
+                actualDefault = field.get(checkInstance);
+                propertyType = field.getType();
+            }
+            final String actualDefaultStr;
+            if (actualDefault == null) {
+                actualDefaultStr = NULL_STRING;
+            }
+            else {
+                actualDefaultStr = convertDefaultValueToString(actualDefault, propertyName);
+            }
+            if (!isDefaultValues(propertyDefaultValue, actualDefaultStr, propertyType)) {
+                throw new IllegalArgumentException("Default value mismatch for " + propertyName
+                        + " in " + fullyQualifiedModuleName + ": specified '" + propertyDefaultValue
+                        + "' but actually is '" + actualDefaultStr + "'");
+            }
+
+        }
+        catch (ReflectiveOperationException ex) {
+
+            hardCodedDefault(propertyName, fullyQualifiedModuleName, propertyDefaultValue);
+
```

Comment:
remove extra empty lines around hardCodedDefault.

it should be
```
actualDefaultStr = getExceptionalDefaults(propertyName, fullyQualifiedModuleName, propertyDefaultValue);
isDefaultValues(propertyDefaultValue, actualDefaultStr, String.class);
```

to clearly see you do validation in "catch"